hooks/context_change.py

```diff
@@ -17,8 +17,14 @@
 
 class ContextChangeHook(get_hook_baseclass()):
     """
-    Hook that gets executed every time there is a context change, once before
-    the change and once after.
+    Hook that gets executed every single time there is a context change in Toolkit,
+    e.g. when an engine starts or restarts and when sgtk.platform.change_context
+    is invoked.
+
+    .. note:
+        If the engine is restarted without changing context or sgtk.platform.change_context
+        is invoked with the same context, the hook will be executed. To detect
+        such situations, simple compare the two contexes.
```

Comment:
I think this could be more clear.  Maybe something like:

```
Note:

These hooks are called whenever the context is being set in Toolkit.  When a context is
first set current_context may be None.  It is possible that the next_context will be the
same as the current_context (for example during an engine reload).  If you want to
trigger some behavior only when the new context is different than the old context then
you'll need to compare the two arguments.
```